iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/provider/ConfigBasedIcebergCatalogConfigProvider.java

```diff
@@ -40,11 +39,12 @@
  * gravitino.iceberg-rest.catalog.hive_proxy.catalog-backend = hive
  * gravitino.iceberg-rest.catalog.hive_proxy.uri = thrift://{host}:{port} ...
  */
-public class ConfigBasedIcebergCatalogWrapperProvider implements IcebergCatalogWrapperProvider {
+public class ConfigBasedIcebergCatalogConfigProvider implements IcebergCatalogConfigProvider {
```

Comment:
The class name is a little tricky, is there a better name for it?

Reply:
`FileBasedIcebergCatalogConfigProvider` or `StaticIcebergCatalogConfigProvider` ?

Reply:
FileBasedXXX and CatalogBasedXXX, how about this?

Reply:
`CatalogBasedCatalogConfigProvider` seems a little odd

Reply:
use `StaticCatalogConfigProvider`  and `DynamicCatalogConfigProvider` , cc @theoryxu